src/utils/workspace.ts

```diff
@@ -80,16 +80,21 @@ const projectOpenBehaviorSetting: string = 'projectOpenBehavior';
 /**
  * If the selected folder is not open in a workspace, open it now. NOTE: This may restart the extension host
  */
-export async function ensureFolderIsOpen(fsPath: string, actionContext: IActionContext): Promise<void> {
+export async function ensureFolderIsOpen(fsPath: string, actionContext: IActionContext, message?: string, allowSubFolder: boolean = false): Promise<void> {
     // tslint:disable-next-line:strict-boolean-expressions
     const openFolders: vscode.WorkspaceFolder[] = vscode.workspace.workspaceFolders || [];
     const folder: vscode.WorkspaceFolder | undefined = openFolders.find((f: vscode.WorkspaceFolder): boolean => {
-        return fsUtils.isPathEqual(f.uri.fsPath, fsPath);
+        return fsUtils.isPathEqual(f.uri.fsPath, fsPath) || (allowSubFolder && fsUtils.isSubpath(f.uri.fsPath, fsPath));
     });
 
     if (folder) {
         actionContext.properties.openBehavior = 'AlreadyOpen';
     } else {
+        if (message) {
+            const open: vscode.MessageItem = { title: localize('open', 'Open Folder') };
```

Comment:
Am I missing where you're checking for 'Open Folder' being selected?

Reply:
Nope. 'Open Folder' is the only option and ext.ui handles the 'cancel' case with a UserCancelledError. I agree it looks a little weird, but it definitely works.

Reply:
Might be worth a comment.